### PR TITLE
Refactor tests related to expiresIn and exp

### PR DESCRIPTION
**Pull request by MitMaro**
Next PR for #492.

This change extracts all tests in the current test files related to `expiresIn` and `exp` into a single test file. It also adds several missing tests.

These tests are almost identical to the tests added in #497. I believe I removed most references to `exp` and `expiresIn` from the current tests. The remaining tests are not related to these tests.

I extracted `base64UrlEncode` into a separate file so it could be shared with the not before tests. This utility could also be easily replaced with [`base64url`](https://github.com/brianloveswords/base64url). If this is desired, I will gladly swap it out.

There was one test for a `expiresInSeconds` options that was deprecated several versions ago. I didn't see the value in this test at this point, as it would be treated like any other invalid option, so it has been removed. I can re-add this test is desired.

Any suggestions of missing tests are more than welcome, and I will gladly add!

### CODE REVIEW

**Comment by ziluvatar** [comment]
LGTM. However, not sure why the coverage is now lower than in master:

This PR:
```
===============================
Statements   : 97.02% ( 228/235 )
Branches     : 96.57% ( 197/204 )
Functions    : 100% ( 23/23 )
Lines        : 97.41% ( 226/232 )
===============================
```

Master:
```
===============================
Statements   : 97.45% ( 229/235 )
Branches     : 97.06% ( 198/204 )
Functions    : 100% ( 23/23 )
Lines        : 97.84% ( 227/232 )
===============================
```

I took a look to the tests and they seemed to be fine. Can you take a look to the reports?

About:
> This utility could also be easily replaced with base64url. If this is desired, I will gladly swap it out.

I'd say to keep the current function instead of adding another dependency.

File: test/exp.test.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  describe('`jwt.sign` "exp" claim validation', function () {
+    [
+      true,
+      false,
+      null,
+      undefined,
+      // TODO should these fail in validation?
+      // -Infinity,
+      // Infinity,
+      // NaN,
```
[the remark this answers is not in the document]
Comment (ziluvatar) [reply test/exp.test.js]: (sorry I forgot to publish this comment ^ )

[the remark this answers is not in the document]
Comment (MitMaro) [reply test/exp.test.js]: Yup, I meant to remove these for now, like I did in the other tests. I will update this evening.

**Comment by MitMaro** [comment]
I will verify when I get home, but I believe the coverage drop is due to the removal of the `expiresInSeconds` test. My guess is that it was the only test that was testing the case of an invalid option being passed. If that is the case, I can re-add the test until another test is added for an invalid option being passed.

**Comment by MitMaro** [comment]
Confirmed that the drop in coverage is from the removal of the `expiresInSeconds` test. I will re-add that test for now.

File: test/exp.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (MitMaro) [reply test/exp.test.js]: Updated in https://github.com/MitMaro/node-jsonwebtoken/commit/874bca2 before squash.

**Comment by MitMaro** [comment]
Fixed the drop in coverage in https://github.com/MitMaro/node-jsonwebtoken/commit/fb405eb before squash.

**Comment by JacoKoster** [comment]
Excellent work, ship it 👍

**Pull request merged**